neuralmonkey/train.py

```diff
@@ -39,25 +39,33 @@ def create_config(config_file):
     config.add_argument('minimize', bool, required=False, default=False)
     config.add_argument('postprocess')
     config.add_argument('name', str)
+    config.add_argument('random_seed', int, required=False)
     config.add_argument('initial_variables', str, required=False, default=[])
     config.add_argument('overwrite_output_dir', bool, required=False,
                         default=False)
 
-    return config.load_file(config_file)
+    return config
 
 def main():
     if len(sys.argv) != 2:
         print("Usage: train.py <ini_file>")
         exit(1)
 
-    # random seeds have to be set before anything is created in the graph
-    random.seed(2574600)
-    np.random.seed(2574600)
-    tf.set_random_seed(2574600)
-
-    args = create_config(sys.argv[1])
-
-    print("")
+    # define valid parameters and defaults
+    cfg = create_config()
+    # load the params from the config file, getting also the simple arguments
+    args = cfg.load_file(sys.argv[1])
+    ## various things like randseed or summarywriter should be set up here
+    ## so that graph building can be recorded
+    # build all the objects specified in the config
+
+    if args.random_seed is None:
+        args.random_seed = 2574600
```

Comment:
Taky nerozumim, proč nepoužije `default`, ale je to ostatně jedno, když to časem stejně zmizí. Klidně bych to všechno už zmergoval, ať je pokoj.